doc/index.md

```diff
@@ -1032,68 +1050,153 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
+
+Create a new instance of an HTTP Server by passing in table `options`:
+
+  - `.client_timeout`: Timeout (in seconds) to wait for client to send first bytes and/or complete TLS handshake (default: 10)
+  - `.cq` (optional): A cqueues controller to use
+  - `.ctx`: An `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
+  - `.family`: (optional) Protocol family
+  - `.host`: Address to bind to (required if not `.path`)
+  - `.port`: Port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
+  - `.path`: Path to UNIX socket (required if not `.host`)
+  - `.v6only`: Allow ipv6 only (no ipv4-mapped-ipv6)
+  - `.mode`: (optional) `fchmod` or `chmod` socket after creating UNIX domain socket
+  - `.mask`: (optional) Set and restore umask when binding UNIX domain socket
+  - `.unlink`: (optional) Unlink socket path before binding
+  - `.max_concurrent`: Maximum number of connections to allow live at a time (default: infinity)
+  - `.onerror`: Function that will be called when an error occurs (default: do nothing). See [server:onerror()](#http.server:onerror)
+  - `.onstream`: Callback function for handling a new client request. The function requires parameters for [*server*](#http.server) and [*stream*](#stream) and if your function throws an error it will be reported from [*step*](#http.server:step) or [*loop*](#http.server:loop)
+  - `.reuseaddr`: (optional) Turn on SO_REUSEADDR flag
+  - `.reuseport`: (optional) Turn on SO_REUSEPORT flag
+  - `.tls`: `nil`: Allow both tls and non-tls connections
+  -         `true`: Allows tls connections only
+  -         `false`: Allows non-tls connections only
+  - `.version`: The http version to allow to connect (default: any)
+
+
+#### Example {#http.server.listen-example}
 
-### `listen(options)` <!-- --> {#http.server.connect}
+```lua
+
+local app_server = http_server.listen {
+		host = 192.168.0.1;
+		port = 80;
+		onstream = function(server, stream)  
+			local request_headers = assert(stream:get_headers())
+			local request_method = request_headers:get ":method"
+
+			print(string.format('[%s] "%s %s HTTP/%g"  "%s" "%s" ',
+				os.date("%d/%b/%Y:%H:%M:%S %z"),
+				request_headers:get(":method") or "",
+				request_headers:get(":path") or "",
+				stream.connection.version,
+				request_headers:get("referer") or "-",
+				request_headers:get("user-agent") or "-"
+				))
+		end;
+}
+
+```
 
 
 ### `server:onerror(new_handler)` <!-- --> {#http.server:onerror}
 
+If called with parameters, the function replaces the current error handler function with `new_handler`  and returns a reference to the old function. Calling the function with no parameters returns the current error handler. The default error handler is to do nothing, and can be set during instantiation through the `options` table passed to the [*server.listen(options)*](#server.listen) function.
```

Comment:
That was quite a while ago. See master now: https://github.com/daurnimator/lua-http/blob/3f1bf0dd5fd1f930a222c2cfabf8d28a652f5619/http/server.lua#L388